Orange/widgets/visualize/owboxplot.py

```diff
@@ -224,14 +232,56 @@ def set_data(self, dataset):
             else:
                 self.group_var = None  # Reset to trigger selection via callback
             self.openContext(self.dataset)
-            self.attr_changed()
+            self.grouping_changed()
         else:
             self.reset_all_data()
 
+    def apply_sorting(self):
+        def compute_score(attr):
+            if attr is group_var:
+                return 3
+            if attr.is_continuous:
+                # One-way ANOVA
+                col = data.get_column_view(attr)[0]
+                groups = (col[group_col == i] for i in range(n_groups))
+                groups = (col[~np.isnan(col)] for col in groups)
+                groups = [group for group in groups if len(group)]
+                return f_oneway(*groups)[1] if groups else 2
```

Comment:
To wet the appetite, in pandas, this could be:
```py
if attr.is_continuous:
    groups = data.groupby(group_var)[attr].apply(list)
    f_oneway(*groups)[1] if len(groups) else 2